### PR TITLE
bug fixes/forms & validation

### PR DESCRIPTION
**Pull request by dav1dchang**
# What's here (& why)

- made some changes to modals/forms so that the validation is fully working and intercepts any attempt at submitting a non-valid form.
- fixed a lot of form bugs, at times, when i would click edit on a form, the fields would not populated. that has been fixed on all the edit forms
- clicking back and forth from edit/create program would not populate/clear the fields as needed. that has been fixed.
- userManagement edit button now populates a modal and you can now edit the role of a user.
- TO-DO: still cannot edit the enrolled/not-enrolled courses of a specific user because the hyperlink needs to be added on the backend.

- (include link(s) to applicable Trello card(s))

# Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete / work-in-progress, PR is for discussion or feedback
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How has this been tested?

- [x] Tested locally
- [ ] Test suite provides the same results as prior to making changes

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: src/components/RestUsers/AddNewUserForm.js [hunk 1]
```diff
@@ -24,9 +25,24 @@ const AddNewUserForm = props => {
...
+  const onSubmit = values => {
+    const { email, username, firstname, lastname, roleType } = values;
+    const newUser = {
+      email,
+      username,
+      firstname,
+      lastname,
+      roleType,
+    };
+    console.log({ values });
```
Comment (rheactdev) [inline src/components/RestUsers/AddNewUserForm.js]: Remove this console.log

File: src/components/RestUsers/AddNewUserForm.js [hunk 2]
```diff
@@ -24,9 +25,24 @@ const AddNewUserForm = props => {
...
+    console.log({ newUser });
```
Comment (rheactdev) [inline src/components/RestUsers/AddNewUserForm.js]: Remove this console.log

**Review by BUGBOUNTYchrisg8691 — APPROVED** [review]
looks good

**Review by rheactdev — APPROVED** [review]
Good to go!!

**Pull request merged**